src/spark-core.js

```diff
@@ -143,7 +143,11 @@ var SparkCore = AmpersandState.extend({
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
       })
-      .then(this.credentials.logout.bind(this.credentials));
+      .then(function cleanUpAndNotify() {
+        this.credentials.logout();
```

Comment:
this function returns a promise, so i think you either want to trigger before calling credentials.logout, or cache its promise, trigger, then return its promise